linera-execution/src/evm/revm.rs

```diff
@@ -856,6 +897,7 @@ where
     }
 
     fn execute_message(&mut self, message: Vec<u8>) -> Result<(), ExecutionError> {
+        self.db.set_contract_address()?;
```

Comment:
Why do we call `set_contract_address` in every function here?

Reply:
This is explained in `linera-execution/src/evm/database.rs`:
```rust
    pub fn new(runtime: Runtime) -> Self {
        let storage_stats = StorageStats::default();
        // We cannot acquire a lock on runtime here.
        // So, we set the contract_address to a default value
        // and update it later.
        Self {
            storage_stats: Arc::new(Mutex::new(storage_stats)),
            contract_address: Address::ZERO,
            runtime: Arc::new(Mutex::new(runtime)),
            changes: HashMap::new(),
	}
    }
```

Reply:
That is the logical design of the scheme.
We don't want a single file.

Reply:
That's not what I was asking about. I asked for the explanation here (or on `set_contract_address` function) explaining why we have to call it in the first place. Do not explain it here in the GH comments but leave that in the code for the future readers.